src/net_processing.cpp

```diff
@@ -2380,6 +2381,15 @@ uint32_t PeerManagerImpl::GetFetchFlags(const Peer& peer) const
     return nFetchFlags;
 }
 
+uint32_t PeerManagerImpl::GetBlockFetchFlags(const Peer& peer, const CBlockIndex* index) const
+{
+    uint32_t fetch_flags = GetFetchFlags(peer);
+    if (m_chainman.m_blockman.IsPruneMode() && WITH_LOCK(m_chainman.GetMutex(), return m_chainman.IsAssumedValid(index))) {
```

Comment:
Can we also check if we are after the first segwit block? The blocks before have no witness data so don't require any more bandwidth, and then peers can still serve directly from disk.

Reply:
Yes I think that would be possible, although I wonder if that should happen on the server side (That is, have the server always serve pre-segwit blocks straight from disk). It doesn't seem like it makes sense to support serving pre-segwit blocks not directly from disk.